src/server/routers/progress.ts

```diff
@@ -1,10 +1,37 @@
 import {router, baseProcedure} from '../trpc'
 import {z} from 'zod'
-import {ACCESS_TOKEN_KEY} from '../../utils/auth'
 import {loadLessonProgress, loadPlaylistProgress} from '../../lib/progress'
 import {loadUserCompletedCourses} from 'lib/users'
 
 export const progressRouter = router({
+  completedCourseIds: baseProcedure.query(async ({ctx}): Promise<number[]> => {
+    const {userId, userToken} = ctx
+    if (!userToken || !userId) return []
+
+    // select id from playlists join series_progresses on series_progresses.progressable_id = playlists.id where series_progresses.user_id = 71775 and series_progresses.last_lesson_watched_at is not null and series_progresses.is_complete = true;
+    const completedCourseIds = (
+      await ctx.prisma.playlist.findMany({
+        select: {
+          id: true,
+        },
+        where: {
+          series_progresses: {
+            some: {
+              progressable_type: 'Playlist',
+              user_id: userId,
+              last_lesson_watched_at: {
+                not: null,
+              },
+              is_complete: true,
+            },
+          },
+        },
+      })
```

Comment:
joins in prisma queries are 'implicit' meaning you set up relationships in the schema and then need to know that you can use those relationships in `where` and `include` clauses